routes/api/users.js

```diff
@@ -71,17 +70,33 @@ router.post("/users/login", function(req, res, next) {
 });
 
 router.post("/users", function(req, res, next) {
-    const user = new User();
+    // const user = new User();
 
-    user.username = req.body.user.username;
-    user.email = req.body.user.email;
-    user.setPassword(req.body.user.password);
+    // user.username = req.body.user.username;
+    // user.email = req.body.user.email;
+    // user.setPassword(req.body.user.password);
+
+    // user.save()
+    //     .then(function() {
+    //         return res.json({ user: user.toAuthJSON() });
+    //     })
+    //     .catch(next);
+
+    const username = req.body.username,
+    email = req.body.email,
+    hash= req.body.password;
+
+    User.create({ username, email, hash})
+    .then(function(user){
+        return res.status(201).json({ 
+            email: user.email,
+            token: 'This feature is yet to be developed',
```

Comment:
@mppdg the former setup only returns the `email` and `username`, how come you're adding more details ? see the return value of `toAuthJson()`